@here/harp-geoutils/lib/projection/MercatorProjection.ts

```diff
@@ -105,6 +105,10 @@ class MercatorProjection extends Projection {
         return geoPoint;
     }
 
+    unprojectAltitude(worldPoint: Vector3Like): number {
+        return worldPoint.z;
```

Comment:
for mercator worldPoint.z is not in meters. You need to apply the mercator scale